src/t_cose_key.c

```diff
@@ -108,3 +110,56 @@ t_cose_key_decode(struct q_useful_buf_c cbor_encoded,
 
     return result;
 }
+
+/*
+ * Public function. See t_cose_key.h
+ */
+enum t_cose_err_t
+t_cose_key_encode(struct t_cose_key      key,
+                  struct q_useful_buf    key_buf,
+                  struct q_useful_buf_c *cbor_encoded)
+{
+    enum t_cose_err_t      result;
+    int32_t                cose_curve;
+    MakeUsefulBufOnStack(  x_coord_buf, T_COSE_BITS_TO_BYTES(T_COSE_ECC_MAX_CURVE_BITS));
+    MakeUsefulBufOnStack(  y_coord_buf, T_COSE_BITS_TO_BYTES(T_COSE_ECC_MAX_CURVE_BITS));
+    struct q_useful_buf_c  x_coord;
+    struct q_useful_buf_c  y_coord;
+    bool                   y_sign;
+    QCBOREncodeContext     cbor_encoder;
+    QCBORError             qcbor_result;
+
+    result = t_cose_crypto_export_ec2_key(key,
+                                          &cose_curve,
+                                          x_coord_buf,
+                                          &x_coord,
+                                          y_coord_buf,
+                                          &y_coord,
+                                          &y_sign);
+    if (result != T_COSE_SUCCESS) {
+        return result;
+    }
+
+    QCBOREncode_Init(&cbor_encoder, key_buf);
+
+    QCBOREncode_OpenMap(&cbor_encoder);
+
+    QCBOREncode_AddInt64ToMapN(&cbor_encoder, T_COSE_KEY_COMMON_KTY, T_COSE_KEY_TYPE_EC2);
+    QCBOREncode_AddInt64ToMapN(&cbor_encoder, T_COSE_KEY_PARAM_CRV, cose_curve);
+    QCBOREncode_AddBytesToMapN(&cbor_encoder, T_COSE_KEY_PARAM_X_COORDINATE, x_coord);
+    if (q_useful_buf_c_is_null(y_coord)) {
```

Comment:
If this function is limited to EC2, both coordinates are needed - i.e., the compressed form x + y-sign is not allowed.